src/Components/WebAssembly/JSInterop/src/InternalCalls.cs

```diff
@@ -3,19 +3,40 @@
 
 using System.Diagnostics.CodeAnalysis;
 using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices.JavaScript;
 
 namespace WebAssembly.JSInterop;
 
 /// <summary>
 /// Methods that map to the functions compiled into the Mono WebAssembly runtime,
 /// as defined by 'mono_add_internal_call' calls in driver.c.
 /// </summary>
-internal static class InternalCalls
+internal static partial class InternalCalls
 {
     // The exact namespace, type, and method names must match the corresponding entries
     // in driver.c in the Mono distribution
     /// See: https://github.com/mono/mono/blob/90574987940959fe386008a850982ea18236a533/sdks/wasm/src/driver.c#L318-L319
 
     [MethodImpl(MethodImplOptions.InternalCall)]
+    [Obsolete]
```

Comment:
Could you please add comment on why we still keep this internal method. 
And also set expectations on how long it would stay with us.